src/test/java/uk/nhs/hee/tis/trainee/sync/migration/CopyProgrammeMembershipToCurriculumMembershipTest.java

```diff
@@ -0,0 +1,136 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.WriteModel;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import org.bson.Document;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.DefaultIndexOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.index.Index;
+import org.springframework.data.mongodb.core.index.IndexField;
+import org.springframework.data.mongodb.core.index.IndexInfo;
+import uk.nhs.hee.tis.trainee.sync.model.CurriculumMembership;
+
+class CopyProgrammeMembershipToCurriculumMembershipTest {
+  private static final String SOURCE_COLLECTION = "programmeMembership";
+  private static final String DEST_COLLECTION = "curriculumMembership";
+  private static final String DEST_CLASS_NAME = CurriculumMembership.class.getName();
+
+  private CopyProgrammeMembershipToCurriculumMembership migration;
+
+  private MongoTemplate template;
+  private DefaultIndexOperations indexOperationsSource;
+  private DefaultIndexOperations indexOperationsDest;
+  private MongoCollection<Document> sourceCollection;
+  private MongoCollection<Document> destCollection;
+
+  @BeforeEach
+  void setUp() {
+    template = mock(MongoTemplate.class);
+    indexOperationsSource = mock(DefaultIndexOperations.class);
+    indexOperationsDest = mock(DefaultIndexOperations.class);
+    sourceCollection = mock(MongoCollection.class);
+    destCollection = mock(MongoCollection.class);
+    migration = new CopyProgrammeMembershipToCurriculumMembership(template);
+  }
+
+  @Test
+  void shouldCreateCurriculumMembershipTable() {
+    IndexField idxFieldSimple = IndexField.create("simpleIndexField", Sort.Direction.ASC);
+    IndexField idxFieldCompound1 = IndexField.create("compoundIndexField1", Sort.Direction.ASC);
+    IndexField idxFieldCompound2 = IndexField.create("compoundIndexField2", Sort.Direction.ASC);
+    IndexInfo idxInfoSimple = new IndexInfo(List.of(idxFieldSimple), "test", false, false, "en");
+    IndexInfo idxInfoCompound = new IndexInfo(List.of(idxFieldCompound1, idxFieldCompound2), "test2", false, false, "en");
+
+    //given
+    when(template.collectionExists(DEST_COLLECTION)).thenReturn(false);
+    when(template.indexOps(SOURCE_COLLECTION)).thenReturn(indexOperationsSource);
```

Comment:
This feels a bit like the test is knowing too much about the inner workings of the method under test, but I'm not sure how else to establish reasonable coverage at this point.